doc/contribute-docs.md

```diff
@@ -1,25 +1,61 @@
 # How to contribute documentation
+This guide aims to help you with all the information necessary to contribute to the Netplan documentation, especially if you're contributing for the first time.
 
 ## Reporting an issue
 
-If you find any issue in Netplan documentation please [file a bugreport](https://bugs.launchpad.net/netplan/+filebug?field.tags=documentation)
-about it in our bug tracker on Launchpad. Remember adding a `documentation` tag
+If you find any issue in Netplan documentation please [file a bug report](https://bugs.launchpad.net/netplan/+filebug?field.tags=documentation) about it in our bug tracker on Launchpad. Remember to add a `documentation`` tag
 to it.
 
 ## Modifying documentation online
 
-Each documentation page rendered on the web contains an **Edit this page** link
-in the top-right of every page. Clicking this button will lead you to the GitHub
+Each documentation page rendered on the web contains an **Edit this **page** link at the top-right of every page. Clicking this button will lead you to the GitHub
 web editor where you can propose changes to the corresponding page.
 
 Please remember to first check the [latest version](https://netplan.readthedocs.io/en/latest/)
 of our documentation and make your proposal based on that revision.
 
-## Creating a pull request
+## Contributing on GitHub
 
-If you want to follow a Git development workflow, you can also checkout the
+If you want to follow a Git development workflow, you can also check out the
 [Netplan repository](https://github.com/canonical/netplan) and contribute your
 changes as [pull requests](https://github.com/canonical/netplan/pulls), putting
 the `documentation` label for better visibility.
 
-See the `doc/` and `examples/` directories for relevant files.
+## Folder Structure
+All the documentation files are located in the `doc/` directory. The `doc/` directory contains sub-directories corresponding to different Diataxis sections. So you should find the following folders inside the `doc/` directory:
+* `tutorial`
+* `explanation`
+* `howto`
+* `reference`
+
+If you're adding a new article, ensure you include it in the appropriate Diataxis directory. You can read about [how Ubuntu implements Diataxis for documentation](https://ubuntu.com/blog/diataxis-a-new-foundation-for-canonical-documentation).
+
+## How to Build the Documentation
+These steps will show you how to build the documentation on your local machine.
+1. To build this documentation, first create a fork of the [Netplan repository](https://github.com/canonical/netplan) and clone that into your machine. For example:
+    ```shell
+    git clone git@github.com:ade555/netplan.git
```

Comment:
Thanks for your contribution. May I ask you to change this to a generic github link instead of your own? Some people tend to copy and paste things verbatim :)

Reply:
> Thanks for your contribution. May I ask you to change this to a generic github link instead of your own? Some people tend to copy and paste things verbatim :)

Definitely. I'll change it right now.